core/src/services/sftp/backend.rs

```diff
@@ -309,9 +281,9 @@ impl Accessor for SftpBackend {
     }
 
     async fn create_dir(&self, path: &str, _: OpCreate) -> Result<RpCreate> {
-        let client = self.sftp_connect().await?;
-        let mut fs = client.sftp.fs();
-        fs.set_cwd(self.root.clone());
+        let client = self.connect().await?;
+        let mut fs = client.fs();
+        fs.set_cwd(&self.root);
 
         let paths: Vec<&str> = path.split_inclusive('/').collect();
```

Comment:
std::path::Path has an associated fn called components, which is more robust than manually parsing path